File: LeetCode/1508-range-sum-of-sorted-subarray-sums/1508-range-sum-of-sorted-subarray-sums.py
class Solution:
    def rangeSum(self, nums: List[int], n: int, left: int, right: int) -> int:
        arr = []
        
        for i in range(n):
            sum = 0
            for j in range(i, n):
                sum += nums[j]
                arr.append(sum)
        
        arr.sort()
        
        MOD = 1_000_000_007
        ans = 0
        for i in range(left - 1, right):
            ans = (ans + arr[i]) % MOD
        
        return ans

File: LeetCode/1508-range-sum-of-sorted-subarray-sums/1508-range-sum-of-sorted-subarray-sums.cpp
class Solution {
public:
    int rangeSum(vector<int>& nums, int n, int left, int right) {
        vector<long long> arr;

        for (int i = 0; i < n; ++i) {
            long long sum = 0;
            for (int j = i; j < n; ++j) {
                sum += nums[j];
                arr.push_back(sum);
            }
        }

        sort(arr.begin(), arr.end());

        const long long mod = 1e9 + 7;
        long long ans = 0;
        for (int i = left - 1; i < right; ++i) {
            ans = (ans + arr[i]) % mod;
        }

        return static_cast<int>(ans);
    }
};